### ISSUE TITLE
[Feature request] Edit prompt with external text editor and log output.

### ISSUE THREAD

**Issue by Disonantemus** [aandrew-me/tgpt#56]
It's nice to use *gpt* in terminal, but the built-in editor is lacking with essential editing, to not write anything again, like this:
- move cursor a character (left, right)
- move cursor a word
- move to line start
- move to line end

---

- If I press: 
`Left Right Up Down Insert Delete Home End PageUp PageDown` 
`Esc F1-12`

- I get **garbage** like this:
`^[[D^[[C^[[A^[[B^[[2~^[[3~^[[1~^[[4~^[[5~^[[6~`
`^[^[OP^[OQ^[OR^[OS^[[15~^[[17~^[[18~^[[19~^[[20~^[[21~^[[23~^[[24~`

Some things work, like:
~~~
^u: delete all to the left
^w: delete previous word
^i: tab
^d^d: EOF
^j: Enter?
^z: job to background
^c: exit
^v: input literal character
^m: Enter?
~~~

1. Because all of this, maybe you can allow to **edit prompts with an external editor** (Like: vi, emacs, nano, etc) as an alternative to the built-in editor.

2. **Log output**: this useful because, if I copy terminal output (from tmux) to a text editor, lines get hard wrapped to the terminal size, and needs to be fixed.

3. Maybe there other ways to fix this or has another useful shortcuts, but don't appear at Man, Help, Readme.md or Github at all.

---

OS: Arch Linux x86_64
Shell: fish 3.6.1
Resolution: 1920x1080
WM: dwm
Terminal: tmux

**Comment by Disonantemus** [comment]
Later, I did find that [#23](https://github.com/aandrew-me/tgpt/issues/23) fix my text editing issues with built-in text editor using **multiline mode**, i've never tried that before, because **interactive mode** was *working*, maybe this should be **default**?.

- Why built-in **interactive mode** don't work like this?

**Comment by juren53** [comment]
In Linux, you can move forward one word at a time with Alt + F and backward
on word at a time with Alt + B.

**Comment by aandrew-me** [comment]
> Later, I did find that [#23](https://github.com/aandrew-me/tgpt/issues/23) fix my text editing issues with built-in text editor using **multiline mode**, i've never tried that before, because **interactive mode** was _working_, maybe this should be **default**?.
> 
> * Why built-in **interactive mode** don't work like this?

Yes I added multiline interactive mode as a replacement to normal interactive mode but I didn't remove the old one because some people may have liked to use that way

**Comment by Disonantemus** [comment]
I did try [juren53](https://github.com/juren53) idea, but didn't work.

Using that:

| Keyboard | I get |
|---|---|
|Alt+b| ^[b |
|Alt+f | ^[f |

Shows that and don't move the cursor. Using ant **Terminal** (tmux, kitty, tty).
Seems a **bug** to me, because ^w and ^u work.
